VMEncryption/main/handle.py

```diff
@@ -797,7 +797,9 @@ def enable():
         encryption_operation = public_settings.get(CommonVariables.EncryptionEncryptionOperationKey)
         check_release_rsa_or_ec_key = False
         if security_Type == CommonVariables.ConfidentialVM:
-            check_release_rsa_or_ec_key = True
+            if CommonVariables.KeyEncryptionKeyURLKey in public_settings:
```

Comment:
are we saying KEK is not required if temp disk encryption is attempted, whereas if data disk encryption is attempted then KEK is required and SKR must be attempted.

Reply:
Correct.

Reply:
currently I do not see the precheck for 
ConfidentialEncryptionDataDisk is true then KEK should be present, 
or we can include this check this condition for SKR. 
if ConfidentialEncryptionDataDisk and KEK 
    do SKR

Reply:
We can add that check for ConfidentialEncryptionDataDisk true scenario. However, note that this flag will go away when data disk enc goes public preview.

Reply:
yes, I agree with your comments.